### PR TITLE
[ADD] edi_account

### PR DESCRIPTION
**Pull request by etobella**
This module modifies account.move and makes it an exchange consumer.

As a first, approach, a listener was added in order to check entrance points for data, but it is open to discussion

@simahawk

### CODE REVIEW

File: edi_account/tests/test_edi.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                            "credit": 1650.0,
+                        },
+                    ),
+                ],
+            }
+        )
+
+    def setUp(self):
+        super().setUp()
+        self._load_module_components("component_event")
```
Comment (simahawk) [inline edi_account/tests/test_edi.py]: you can do this in setUpClass. BTW could you review this? https://github.com/OCA/connector/pull/387

File: edi_account/tests/test_edi.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            # _apply_on = ["account.move"]
+
+            def on_post_account_move(self, move):
+                move.name = "new_name"
+
+        MyEventListener._build_component(self.comp_registry)
+        self.test_move.post()
+        # As we are testing, the component is not executed properly, this way we can
+        # check that the component event will be called
+        events = self.test_move._event(
```
Comment (simahawk) [inline edi_account/tests/test_edi.py]: w/ the PR linked above you don't need anymore to pass the comp registry

File: edi_account/tests/test_edi.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+from odoo.addons.component.core import Component
+from odoo.addons.component.tests.common import SavepointComponentRegistryCase
+
+_logger = logging.getLogger(__name__)
+
+
+class EDIBackendTestCase(SavepointComponentRegistryCase, AccountTestInvoicingCommon):
+    @classmethod
+    def setUpClass(cls, chart_template_ref=None):
+        super().setUpClass(chart_template_ref=chart_template_ref)
```
Comment (simahawk) [inline edi_account/tests/test_edi.py:18]: pls set tracking_disable suite wise

**Review by simahawk — CHANGES_REQUESTED** [review]
LG as a base

File: edi_account/tests/test_edi.py (hunk 1 above)
Comment (etobella) [reply edi_account/tests/test_edi.py]: I tried to do it on setUpClass and an error was raised :cry: I can try to find why

Comment (simahawk) [reply edi_account/tests/test_edi.py]: https://github.com/OCA/edi/pull/264/files#diff-1917008f7c39079edbeb6a81a96e98304bff5e2d327ea0443a960a3eed4567ccR90 :wink:

File: edi_account/tests/test_edi.py (hunk 3 above)
Comment (etobella) [reply edi_account/tests/test_edi.py:18]: done

**Review by jarroyomorales — APPROVED** [review]
Code review LGTM :+1:

File: edi_account/models/account_move.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        result = super().post()
+        # We will use this event to know which documents needs to be executed
+        if self:
+            self._event("on_post_account_move").notify(self)
+        return result
+
+    def button_cancel(self):
+        """This could be used to notify our provider that we are not accepting the
+        invoice"""
+        result = super(AccountMove, self).button_cancel()
```
Comment (bealdav) [inline edi_account/models/account_move.py]:
```suggestion
        result = super().button_cancel()
```

File: edi_account/readme/DESCRIPTION.rst [hunk 5]
```diff
@@ -0,0 +1,16 @@
+This module intends to create a base to be extended by local edi rules
+for accounting.
+
+In order to add a new integracion for an account, you need to create a listener:
```
Comment (bealdav) [inline edi_account/readme/DESCRIPTION.rst]:
```suggestion
In order to add a new integration for an account, you need to create a listener:
```

**Comment by etobella** [comment]
@bealdav comments attended

**Review by simahawk — APPROVED** [review]

**Pull request merged**